bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ErrorValue.java

```diff
@@ -302,9 +302,20 @@ public String getPrintableStackTrace() {
         for (int i = 1; i < stackTrace.length; i++) {
             printStackElement(sb, stackTrace[i], "\n\t   ");
         }
+        addErrorCauseWithLocation(sb, this.cause);
         return sb.toString();
     }
 
+    public void addErrorCauseWithLocation(StringBuilder sb, BError cause) {
+        if (cause != null) {
+            sb.append("\ncause: ")
+                    .append(cause.getMessage())
+                    .append("\n\tat ");
+            printStackElement(sb, cause.getStackTrace()[0], "");
```

Comment:
can there be any cases where `cause.getStackTrace()` return an empty array?

Reply:
When the cause is not `null` we will always have elements in `StackTraceElement[]`. So IMO, do not need to have a separate check for that.